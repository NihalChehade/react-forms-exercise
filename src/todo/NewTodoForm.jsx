import React, {useState} from 'react'

const NewTodoForm = ({addTodo}) => {
  const [todo, setTodo] = useState('');

  const handleChange = (e)=>{
   
    setTodo(e.target.value)
  }
  const handleSubmit = (e)=>{
    e.preventDefault();
    addTodo(todo);
    setTodo('')
  }
  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="todo">Todo:</label>
      <input 
      type="text" 
      name="todo" 
      id="todo" 
      value={todo} 
      onChange={handleChange}/>
      <button type="submit">Add Todo!</button>
    </form>
  )
}

export default NewTodoForm