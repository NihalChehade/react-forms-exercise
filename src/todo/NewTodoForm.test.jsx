import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

// Smoke Test
test('renders NewTodoForm component without crashing', () => {
  render(<NewTodoForm addTodo={() => {}} />);
});

// Snapshot Test
test('matches snapshot', () => {
  const { asFragment } = render(<NewTodoForm addTodo={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});

// Critical Business Logic Tests
test('updates state when input changes', () => {
  render(<NewTodoForm addTodo={() => {}} />);
  
  const input = screen.getByLabelText('Todo:');
  fireEvent.change(input, { target: { value: 'New Task' } });
  
  expect(input.value).toBe('New Task');
});

test('calls addTodo with correct data and resets form on submit', () => {
  const mockAddTodo = vi.fn();
  render(<NewTodoForm addTodo={mockAddTodo} />);
  
  const input = screen.getByLabelText('Todo:');
  fireEvent.change(input, { target: { value: 'New Task' } });
  fireEvent.click(screen.getByText('Add Todo!'));
  
  expect(mockAddTodo).toHaveBeenCalledWith('New Task');
  expect(input.value).toBe('');
});
