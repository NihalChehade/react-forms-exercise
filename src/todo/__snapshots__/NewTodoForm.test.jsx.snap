// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todo"
    >
      Todo:
    </label>
    <input
      id="todo"
      name="todo"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Todo!
    </button>
  </form>
</DocumentFragment>
`;
